tests/functional/test_cache.py

```diff
@@ -332,34 +335,97 @@ def test_cache_remove_too_many_args(script: PipTestEnvironment) -> None:
     script.pip("cache", "remove", "aaa", "bbb", expect_error=True)
 
 
+@pytest.mark.parametrize(
+    "pattern",
+    [
+        "zzz",
+        "ZZZ",
+        "zZz-*-py3-none-any.whl",
+        "zZz-*.whl",
+        "zzz-*",
+        "zzz*",
+        "zz[a-zA-Z]?",
+        "[!y]",
+    ],
+)
 @pytest.mark.usefixtures("populate_wheel_cache")
 def test_cache_remove_name_match(
-    script: PipTestEnvironment, remove_matches_wheel: RemoveMatches
+    script: PipTestEnvironment, pattern: str, remove_matches_wheel: RemoveMatches
 ) -> None:
-    """Running `pip cache remove zzz` should remove zzz-4.5.6 and zzz-7.8.9,
-    but nothing else."""
-    result = script.pip("cache", "remove", "zzz", "--verbose")
+    """Running `pip cache remove <zzz matching pattern>` should remove zzz-4.5.6,
+    zzz-4.5.7 and zzz-7.8.9, but nothing else."""
+    result = script.pip("cache", "remove", pattern, "--verbose")
 
     assert not remove_matches_wheel("yyy-1.2.3", result)
     assert remove_matches_wheel("zzz-4.5.6", result)
     assert remove_matches_wheel("zzz-4.5.7", result)
     assert remove_matches_wheel("zzz-7.8.9", result)
 
 
+@pytest.mark.parametrize(
+    "pattern",
+    [
+        "zzz-4.5.6",
+        "ZZZ-4.5.6",
+        "zZz-4.5.6-py3-none-any.whl",
+        "zZz-4.5.6-*.whl",
+        "zZz-4.5.[0-6]",
+    ],
+)
 @pytest.mark.usefixtures("populate_wheel_cache")
 def test_cache_remove_name_and_version_match(
-    script: PipTestEnvironment, remove_matches_wheel: RemoveMatches
+    script: PipTestEnvironment, pattern: str, remove_matches_wheel: RemoveMatches
 ) -> None:
-    """Running `pip cache remove zzz-4.5.6` should remove zzz-4.5.6, but
-    nothing else."""
-    result = script.pip("cache", "remove", "zzz-4.5.6", "--verbose")
+    """Running `pip cache remove <zzz-4.5.6 matching pattern>` should remove zzz-4.5.6,
+    but nothing else."""
+    result = script.pip("cache", "remove", pattern, "--verbose")
 
     assert not remove_matches_wheel("yyy-1.2.3", result)
     assert remove_matches_wheel("zzz-4.5.6", result)
     assert not remove_matches_wheel("zzz-4.5.7", result)
     assert not remove_matches_wheel("zzz-7.8.9", result)
 
 
+@pytest.mark.parametrize(
```

Comment:
Could you also include some test cases with periods and runs of underscores? The latter is not something I'd expect an user to pass to pip, but it'd be good to validate this is following the specifications.